lib/AgentSDK.js

```diff
@@ -229,8 +229,8 @@ class SDK extends Events {
             let key = utils.toFuncName(reqType);
 
             // create it as a function
-            this[key] = (body, headers, metadata, encodedMetadata, callback) => {
-                this._createAndSendRequest(reqType, body, headers, metadata, encodedMetadata, callback);
+            this[key] = (body, headers, metadata, encodedMetadata, messageAudience, callback) => {
```

Comment:
is it better to move all this arguments to a object? like: 
this[key] = ({body, headers, metadata, encodedMetadata, messageAudience, callback}) => {

it will decrease the chance of error for anyone who is calling it

Reply:
Thanks, I really wanted to do it as you stated because I also didn't like that practice of keep adding the params but I have to keep in mind that all people using this package will be affected. If I get green from package owner I would love to change it as you mentioned

Reply:
I agree with @lmarchel13, moving the arguments to an object decreases the need for many params.

Reply:
@map34 I agree as well but this will not a minor change rather you guys will need to release a major version of AgntSDK and I think such change shouldn't be the part of this feature PR. Moreover, it introduces more implementation time on our side as well so, for now, I would like to keep it this way. I will introduce the unit test soon for this change.

Reply:
I don't understand why an extra argument is needed here. According to https://confluence.liveperson.com/display/SA/Private+Messages "messageAudience" should be an attribute on body, which already has an argument, right?

Reply:
Looks like this also a breaking change, could we also update the documentation and README.md? Also, could we at least make a default value with this to make sure the function is still backward compatible?

Reply:
@map34 & @asicath  Thanks for pointing that out I tested and it seems like we don't need this update and we will send the attribute with the body